src/core/cuda_event.cpp

```diff
@@ -0,0 +1,123 @@
+#include "cuda_event.h"
+
+
+
+using namespace CUDA;
+
+
+
+
+
+
+Event::Event()
+{
+   CUDA_SAFE_CALL(cuEventCreate(&_id, CU_EVENT_DEFAULT));
+}
+
+
+
+
+
+
+Event::Event(Event&& other):
+   _id(other._id)
+{
+   other._id = nullptr;
+}
+
+
+
+
+
+
+Event::~Event()
+{
+   CUDA_SAFE_CALL(cuEventDestroy(_id));
+}
+
+
+
+
+
+
+Event& Event::operator=(Event&& other)
+{
+   std::swap(_id, other._id);
+   return *this;
+}
+
+
+
+
+
+
+/*!
+ * Record the contents of a CUDA stream into a CUDA event.
+ *
+ * @param stream
+ */
+void Event::record(const Stream& stream)
+{
+   CUDA_SAFE_CALL(cuEventRecord(_id, stream.id()));
+}
+
+
+
+
+
+
+/*!
+ * Waits for this object's CUDA event to complete, blocking until it does. If
+ * this event is null then this returns immediately.
+ */
+void Event::wait() const
+{
+   CUDA_SAFE_CALL(cuEventSynchronize(_id));
+}
+
+
+
+
+
+
+/*!
+ * Tests if this object's CUDA event is complete.
+ *
+ * @return True if this event is complete or null, otherwise false.
+ */
+bool Event::isDone() const
+{
+   // query event status
```

Comment:
Severe lack of inline comments, again complete sentences. Also the name of the variable 'error' is extremely misleading because you are not just checking the return value of the query for an error.